### ISSUE TITLE
unexpected behaviour for executing command in thread, doesn't die 

### ISSUE THREAD

**Issue by despiegk** [vlang/v#6929]
```vlang
module mymodule
import sync
import time
import log
import os

fn task(id, duration int, mut wg sync.WaitGroup) {
    mut l := log.Log{}
    l.set_level(.debug)
    //l.fatal("this is fatal error1")
    l.info("debug message ${id}")
    println("task ${id} begin")
    time.sleep_ms(duration)
    println("task ${id} end")
    // next directory does not exist
    e := os.exec("ls /s") or { 
        l.fatal("this is fatal error for list")
        // does not print the fatal error ???
        panic(err) 
        // did not stop the program
    }
    println(e.output)
    wg.done()
}

pub fn say_hi() {
    // ls('/s') or { panic(err) }
    mut wg := sync.new_waitgroup()
    wg.add(3)
    go task(1, 500, mut wg)
    go task(2, 900, mut wg)
    go task(3, 100, mut wg)
    wg.wait()
    println('done')
}
```

next script calls it

```vlang
import mymodule
mymodule.say_hi()
```

```bash
kristofs-MacBook-Pro:v despiegk$ v run test.v
[INFO  2020-11-24 07:37:09] debug message 2
task 2 begin
[INFO  2020-11-24 07:37:09] debug message 3
task 3 begin
[INFO  2020-11-24 07:37:09] debug message 1
task 1 begin
task 3 end
ls: /s: No such file or directory

task 1 end
ls: /s: No such file or directory

task 2 end
ls: /s: No such file or directory

done
```

it does not raise error and die

but if I put the error1 to work it does die

```
kristofs-MacBook-Pro:v despiegk$ v run test.v
[FATAL 2020-11-24 07:39:47] this is fatal error1
V panic: : this is fatal error1
0   test                                0x0000000109b1f989 log__Log_fatal + 137
1   test                                0x0000000109afcf8c mymodule__task + 156
[FATAL 2020-11-24 07:39:47] this is fatal error1
2   test                                0x0000000109afcedc mymodule__task_thread_wrapper + 44
3   libsystem_pthread.dylib             0x00007fff71877109 _pthread_start + 148
4   libsystem_pthread.dylib             0x00007fff71872b8b thread_start + 15
[FATAL 2020-11-24 07:39:47] this is fatal error1
V panic: : this is fatal error1
Terminated by signal 11 (SIGSEGV)
```

**Comment by spytheman** [comment]
After a call to os.exec, you need to check e.exit_code too.

The or {} block in the case of os.exec, is executed when the command was not found.
In the example program, the `ls` executable is found and it is executed, but it itself returns an exit_code of 2, and an error output of `ls: /s: No such file or directory` .

Most unix programs by convention have an exit code of 0, when they are successful.

**Comment by spytheman** [comment]
tldr: try changing `println(e.output)` to `println(e)` to see what is happening.

**Comment by despiegk** [comment]
thank you so much, sorry for asking
